File: app/src/main/java/com/example/javakotlin/javakotlinoverview/java/Inheritance.java
package com.example.javakotlin.javakotlinoverview.java;

class Person {
    String name;
    int age;
}

class Student extends Person {
    String course;
    float fees;
}

class WorkingMan extends Person {
    String occupation;
    float income;
}

public class Inheritance {
    public static void main(String args[]) {
        Student s = new Student();
        s.name = "Bansi";
        s.course = "BTech";

        WorkingMan w = new WorkingMan();
        w.name = "ABC";
        w.occupation = "Professor";

        System.out.println("--Student--");
        System.out.println("name: " + s.name);
        System.out.println("course: " + s.course);

        System.out.println("--WorkingMan--");
        System.out.println("name: " + w.name);
        System.out.println("occupation: " + w.occupation);
    }
}

File: app/src/main/java/com/example/javakotlin/javakotlinoverview/kotlin/Inheritance.kt
package com.example.javakotlin.javakotlinoverview.kotlin

open class Person(name: String,age: Int) {
    init {
        println("My name is $name.")
        println("My age is $age")
    }
}

class Student(name: String,age: Int): Person(name,age) {
    fun getCourse() {
        println("My course is btech")
    }
}

class WorkingMan(name: String,age: Int): Person(name,age) {
    fun getOccupation() {
        println("I work as a salesman")
    }
}
class Inheritance {
    fun main(args: Array<String>) {
        val s = Student("Bansi",20)
        s.getCourse()

        println()

        val w = WorkingMan("ABC", 40)
        w.getOccupation()
    }
}
